pkg/ooniengine/geoip.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+//
+// GeoIP task
+//
+
+import (
+	"context"
+
+	"github.com/ooni/probe-cli/v3/pkg/ooniengine/abi"
+	"google.golang.org/protobuf/proto"
+)
+
+func init() {
+	taskRegistry["GeoIP"] = newGeoIPRunner()
+}
+
+// newGeoIPRunner creates a new geoIPRunner.
+func newGeoIPRunner() taskRunner {
+	return &geoIPRunner{}
+}
+
+// geoIPRunner is the geoip task runner.
+type geoIPRunner struct{}
+
+var _ taskRunner = &geoIPRunner{}
+
+// main implements taskRunner.main.
+func (r *geoIPRunner) main(ctx context.Context, emitter taskMaybeEmitter, args []byte) {
+	logger := newTaskLogger(emitter)
+	var config abi.GeoIPConfig
```

Comment:
pretty obvious, but it seems like there's some boilerplate here in the session init that can be factored out :)